packages/phishing-controller/src/PhishingController.test.ts

```diff
@@ -69,6 +69,56 @@ describe('PhishingController', () => {
     expect(nockScope.isDone()).toBe(false);
   });
 
+  describe('maybeUpdatePhishingLists', () => {
+    it('should not be out of date immediately after maybeUpdatePhishingLists is called', async () => {
+      const clock = sinon.useFakeTimers();
+      const controller = new PhishingController({ refreshInterval: 10 });
+      clock.tick(10);
+      expect(controller.isOutOfDate()).toBe(true);
+      await controller.maybeUpdatePhishingLists();
+      expect(controller.isOutOfDate()).toBe(false);
+    });
+
+    it('should not be out of date after maybeUpdatePhishingLists is called but before refresh interval has passed', async () => {
+      const clock = sinon.useFakeTimers();
+      const controller = new PhishingController({ refreshInterval: 10 });
+      clock.tick(10);
+      expect(controller.isOutOfDate()).toBe(true);
+      await controller.maybeUpdatePhishingLists();
+      clock.tick(5);
+      expect(controller.isOutOfDate()).toBe(false);
+    });
+
+    it('should still be out of date while update is in progress', async () => {
+      const clock = sinon.useFakeTimers();
+      const controller = new PhishingController({ refreshInterval: 10 });
+      clock.tick(10);
+      await controller.maybeUpdatePhishingLists();
+      expect(controller.isOutOfDate()).toBe(false);
+      clock.tick(10);
+      expect(controller.isOutOfDate()).toBe(true);
+      // do not wait
+      controller.maybeUpdatePhishingLists();
+      expect(controller.isOutOfDate()).toBe(true);
+      await controller.maybeUpdatePhishingLists();
+      expect(controller.isOutOfDate()).toBe(false);
+      clock.tick(10);
+      expect(controller.isOutOfDate()).toBe(true);
+    });
+
+    it('should not call update if it is out of date', async () => {
+      const clock = sinon.useFakeTimers();
+      const controller = new PhishingController({ refreshInterval: 10 });
+      const updateSpy = sinon.spy(controller, 'updatePhishingLists');
```

Comment:
Rather than mocking a controller method, could you update this to use `nock` to setup a fetch mock? You can copy one of the other nock scopes used in other tests. This is preferred over mocking functions because `updatePhishingLists` is _part of_ `maybeUpdatePhishingLists`, and is also under test here. By mocking it we might be missing mistakes in how that internal function is used, and we're setting this test up as an obstacle to refactoring this in the future.

Reply:
This is also generating additional console errors when these tests are run. I've just noticed that some pre-existing tests are doing this as well. That's not ideal - I can look at fixing that in a separate PR.

Reply:
Thanks, this looks great!